### DISCUSSION TITLE
How can I know field type by json schema?

### DISCUSSION THREAD

**Discussion by hyeongguen-song** [pydantic/pydantic#7010] [Question]
I used `BaseModel.schema()` for get type and generate raw query.
```py
    def _to_sql_type(self, field: str) -> str:
        field_info = self.schema()["properties"][field]
        field_type = field_info["type"]
        if field_type == "string":
            format_ = field_info.get("format")
            ....
            if format_ == "json-string":
                return "::jsonb"
        ....
```
And also when use `v2.0.3` I can get even if type is `Pydantic.Json`

But I can't get type of Json..(`v2.1.1`) How can I know it?

**Comment by Viicos** [comment]
You could play with `BaseModel.model_fields` instead, which will give you a Pythonic model definition
